File: src/main/java/io/beaniejoy/dongnecafe/cafe/dto/menu/CafeMenuDetailResponseDto.java
package io.beaniejoy.dongnecafe.cafe.dto.menu;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CafeMenuDetailResponseDto {

    private String name;

    private Integer price;

    private List<MenuOptionResponseDto> optionList;
}

File: src/main/java/io/beaniejoy/dongnecafe/domain/cafe/dto/menu/CafeMenuDetailResponseDto.kt
package io.beaniejoy.dongnecafe.domain.cafe.dto.menu

import io.beaniejoy.dongnecafe.domain.cafe.entity.CafeMenu
import java.math.BigDecimal

data class CafeMenuDetailResponseDto(
    val name: String? = null,
    val price: BigDecimal = BigDecimal.ZERO,
    val optionList: List<MenuOptionResponseDto> = emptyList()
) {
    companion object {
        fun of(cafeMenu: CafeMenu): CafeMenuDetailResponseDto {
            return CafeMenuDetailResponseDto(
                name = cafeMenu.name,
                price = cafeMenu.price,
                optionList = cafeMenu.menuOptionList.map { MenuOptionResponseDto.of(it) }
            )
        }
    }
}